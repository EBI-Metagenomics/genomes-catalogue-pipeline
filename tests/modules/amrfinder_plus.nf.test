nextflow_process {

    name "Test Process AMRFINDER_PLUS"
    script "modules/amrfinder_plus.nf"
    process "AMRFINDER_PLUS"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
                amrfinder_plus_db = "$baseDir/ref-dbs/amrfinder_plus/2023-02-23.1"
            }
            process {
                """
                input[0] = channel.from([
                    tuple(
                        "MGYG000299400",
                        file("$baseDir/tests/modules/fixtures/amrfinder_plus/MGYG000299400.fna"),
                        file("$baseDir/tests/modules/fixtures/amrfinder_plus/MGYG000299400.faa"),
                        file("$baseDir/tests/modules/fixtures/amrfinder_plus/MGYG000299400.gff")
                    )
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}
