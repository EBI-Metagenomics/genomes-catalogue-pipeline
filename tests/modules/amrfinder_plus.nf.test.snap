{
    "Should run without failures": {
        "content": [
            {
                "amrfinder_tsv": [
                    [
                        "MGYG000299400",
                        "MGYG000299400_amrfinderplus.tsv:md5,1c60e636f614bab3b200b13c5bd556d8"
                    ]
                ]
            }
        ],
        "timestamp": "2023-03-23T15:00:41+0000"
    }
}
